modules/nf-core/modules/ensemblvep/main.nf

```diff
@@ -16,32 +16,36 @@ process ENSEMBLVEP {
     path  extra_files
 
     output:
-    tuple val(meta), path("*.ann.vcf"), emit: vcf
-    path "*.summary.html"             , emit: report
-    path "versions.yml"               , emit: versions
+    tuple val(meta), path("*.ann.vcf")  , optional:true, emit: vcf
+    tuple val(meta), path("*.ann.tab")  , optional:true, emit: tab
+    tuple val(meta), path("*.ann.json") , optional:true, emit: json
+    path "*.summary.html"               , emit: report
+    path "versions.yml"                 , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
+    def vep_output = task.ext.vep_output ?: 'vcf'
+    def suffix = task.ext.suffix ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     def dir_cache = cache ? "\${PWD}/${cache}" : "/.vep"
     """
     mkdir $prefix
 
     vep \\
         -i $vcf \\
-        -o ${prefix}.ann.vcf \\
+        -o ${prefix}${suffix}.ann.$vep_output \\
```

Comment:
I would propose using the complete output part only defined by `prefix` (this is done in quite a few other modules that have this type of optional output. Then above you could send a default output:

def prefix = task.ext.prefix ?: "${meta.id}.vcf"

Reply:
Right, I'll try to see if I can get that working.

Reply:
So I'm trying to define the `prefix` in the usual way in `modules/nf-core/modules/ensemblvep/main.nf`:
```
def prefix = task.ext.prefix ?: "${meta.id}"
```
And I'm defining `prefix` in `conf/modules.config` like this:
```
    withName: 'NFCORE_SAREK:SAREK:ANNOTATE:ANNOTATION_ENSEMBLVEP:ENSEMBLVEP' {
        ext.prefix       = {"${meta.id}_VEP.ann.${params.vep_out_format}"}
    }
```
Unfortunately, the `prefix` is also used here:
https://github.com/nf-core/sarek/blob/20e28724650c81dfc6aaf7ee6ebdbd9d02e481c7/modules/nf-core/modules/ensemblvep/main.nf#L44
so - if I don't do anything about that - then the summary.html-file would end up with a name like
```
${meta.id}_VEP.ann.${params.vep_out_format}.summary.html
```
But maybe that is okay?

Reply:
what about using `$args2` for the extension (vcf/json/tab)?

Reply:
> what about using `$args2` for the extension (vcf/json/tab)?

Sure, that works. That is similar to my first solution where I put the extension in `task.ext.vep_output`.  Anyways, along with Friederikes suggestions, I think that we now have a good solution. Thanks for all your input.